TF_Settings_Web/src/Components/Controls/TextEntry.tsx

```diff
@@ -0,0 +1,29 @@
+import React, { ChangeEventHandler, MouseEventHandler, PointerEventHandler } from "react";
+
+import '../../Styles/Controls/TextEntry.css';
+
+interface TextEntryProps {
+    name: string,
+    value: number,
+    selected: boolean,
+    onChange: (ChangeEventHandler<HTMLInputElement>),
+    onClick: (MouseEventHandler<HTMLElement>),
+    onPointerDown: (PointerEventHandler<HTMLElement>)
+}
+
+export class TextEntry extends React.Component<TextEntryProps, {}> {
+    render() {
+        return(
+            <label onClick={this.props.onClick.bind(this)} onPointerDown={this.props.onPointerDown.bind(this)} className={"textEntryBackgroundLabel " + (this.props.selected ? "textEntryBackgroundLabelSelected" : "")}>
```

Comment:
Could we move the logic in classname into the body of the `render` function and split this line by each prop? It's a little hard to read at this length.